File: mediumSpringWebflux/src/main/java/com/example/mediumspringwebflux/domain/user/service/UserFacade.java
package com.example.mediumspringwebflux.domain.user.service;

import com.example.mediumspringwebflux.domain.user.document.User;
import com.example.mediumspringwebflux.domain.user.repository.UserRepository;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.ReactiveSecurityContextHolder;
import org.springframework.security.core.context.SecurityContext;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@Component
public class UserFacade {
    private final UserRepository userRepository;

    public UserFacade(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public Mono<User> getUser() {
        return ReactiveSecurityContextHolder.getContext()
                .map(SecurityContext::getAuthentication)
                .flatMap(authentication -> {
                    String email = authentication.getName();
                    return userRepository.findByEmail(email);
                });
    }



}

File: mediumSpringWebMVC/src/main/kotlin/com/example/mediumspringwebmvc/domain/user/service/facade/UserFacade.kt
package com.example.mediumspringwebmvc.domain.user.service.facade

import com.example.mediumspringwebmvc.domain.user.model.User
import com.example.mediumspringwebmvc.domain.user.repository.UserRepository
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class UserFacade(
    private val userRepository: UserRepository
) {

    fun getUser(): User {
        val authentication = SecurityContextHolder.getContext().authentication
        val email = authentication.name

        return userRepository.findByEmail(email) ?: throw RuntimeException("유저를 찾지 못함")
    }

}